UnityProject/Assets/Scripts/Objects/ObjectContainer.cs

```diff
@@ -130,18 +130,23 @@ public void StoreObjects(IEnumerable<GameObject> gameObjects)
 			}
 		}
 
-		public void GatherObjects()
+		//Tries to gather object and checks if closet can be closed.
+		public bool GatherObjects()
 		{
 			foreach (var entity in registerTile.Matrix.Get<ObjectBehaviour>(registerTile.LocalPositionServer, true))
 			{
 				// Don't add the container to itself...
 				if (entity.gameObject == gameObject) continue;
 
+				// We don't want to have locker/crate inside of another locker.
+				if (entity.TryGetComponent<ClosetControl>(out _) == true) return false;
+
 				// Can't store secured objects (exclude this check on mobs as e.g. magboots set pushable false)
-				if (entity.TryGetComponent<HealthV2.LivingHealthMasterBase>(out _) == false && entity.IsPushable == false) continue;
+				if (entity.TryGetComponent<HealthV2.LivingHealthMasterBase>(out _) == false && entity.IsPushable == false) return false;
```

Comment:
wouldn't this stop any object from being picked up if there is a  player with mag boots on?

Reply:
> wouldn't this stop any object from being picked up if there is a player with mag boots on?

Doesnt prevent magboots, the tryGetComponent is looking for false. This would stop closing for secured objects.

Reply:
I dont even see how this connected with picking up items honestly, so prob no

Reply:
what do you think about adding a text thing when this happens?

Reply:
I tried but it caused shitcode. And I dont really think its needed here.

Reply:
I haven't gone into details, but I want to ask a question.

Do I understand correctly that we can't close a locker if it has unstorable objects under it? But we store objects that were checked before the unstorable object?

Shouldn't it be "we store any storable objects. Unstorable objects are ignored"?

Reply:
We cant close locker at all if theres any unstorable objects inside it. If not, we close the locker. In previous version, unstorable objects were just ignored, laying on top of locker but it looked pretty strange.